Firmware/mmu.cpp

```diff
@@ -111,10 +112,25 @@ void mmu_init(void)
 	_delay_ms(10);                             //wait 10ms for sure
 	mmu_reset();                               //reset mmu (HW or SW), do not wait for response
 	mmu_state = -1;
-#ifdef MMU_IDLER_SENSOR_PIN
 	PIN_INP(MMU_IDLER_SENSOR_PIN); //input mode
 	PIN_SET(MMU_IDLER_SENSOR_PIN); //pullup
-#endif //MMU_IDLER_SENSOR_PIN
+}
+
+//returns true if idler IR sensor was detected, otherwise returns false
+bool check_for_idler_sensor() 
+{
+	bool detected = false;
+	//if MMU_IDLER_SENSOR_PIN input is low and pat9125sensor is not present we detected idler sensor
+	if ((PIN_GET(MMU_IDLER_SENSOR_PIN) == 0) && fsensor_not_responding) 
+	{
+		  detected = true;
+		  //printf_P(PSTR("Idler IR sensor detected\n"));
+	}
+	else
+	{
+		  //printf_P(PSTR("Idler IR sensor not detected\n"));
+	}
+	return detected;
 }
 
 //mmu main loop - state machine processing
```

Comment:
Function mmu_loop(void) is too long (over 200 lines of code) hindering review and maintenance effort. mmu_state uses magic numbers instead of symbolic names. Consider enum class for states.

Reply:
mmu_loop() is undocumented.

Reply:
There is no need to specify void function parameter in cpp.